File: backend/api/src/main/java/com/github/ai/leetcodequiz/api/ProblemsItemDto.java
package com.github.ai.leetcodequiz.api;

import org.jetbrains.annotations.NotNull;

public record ProblemsItemDto(
    int id,
    @NotNull String title,
    @NotNull String categoryTitle,
    @NotNull String difficulty,
    @NotNull String url,
    int dislikes,
    int likes
) {
}

File: backend/api/src/main/scala/com/github/ai/leetcodequiz/api/ProblemsItemDto.scala
package com.github.ai.leetcodequiz.api

import zio.json.{JsonDecoder, JsonEncoder}

case class ProblemsItemDto(
  id: Int,
  title: String,
  categoryTitle: String,
  difficulty: String,
  url: String,
  dislikes: Int,
  likes: Int
) derives JsonEncoder, JsonDecoder
